src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,44 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkAge(user);
+        checkPassword(user);
+        checkLogin(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkPassword(User user) {
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Incorrect password!");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Incorrect password!");
+        }
+    }
+
+    private void checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Incorrect user login!");
+        }
+        for (User person : Storage.people) {
```

Comment:
<img width="804" alt="image" src="https://user-images.githubusercontent.com/81781479/218997127-901f9043-07a2-4930-afa3-f344288bf333.png">
use get method to check if the user is present

Reply:
not fixed

Reply:
not fixed

Reply:
not fixed. You have `get` method  in Dao, you don't need a loop here, don't contact storage directly, only via DAO

Reply:
not fixed. You have get method in Dao, you don't need a loop here, don't contact storage directly, only via DAO